### PR TITLE
#169113466 - Edit and Delete Accommodation Facilities

### PR DESCRIPTION
**Pull request by nignanthomas**
#### What does this PR do?
Edit and Delete Accommodation Facilities

#### Description of Task to be completed?
- Create tests cases for edit an accommodation and delete an accommodation features
- Create verifyToken, checkId, isAccommodationFound, isOwner middlewares to validate informations
- Create editAccommodation and deleteAccommodation controllers

#### How should this be manually tested?
- Clone the repository from [here](https://github.com/andela/caret-bn-backend.git)
- Browse to the repo directory
- navigate to `ft-edit-delete-accommodation-169113466` branch
- Run `npm install`
- Refer to .env.example and add `CLOUDINARY_URL=`
- Go to the following url https://developers.google.com/oauthplayground/ and authorize the following APIs (This is done in step 1 of the Playground process).
 a) Google OAuth apiv2
      - https://www.googleapis.com/auth/userinfo.email
      - https://www.googleapis.com/auth/userinfo.profile
- Exchange the authorization code for tokens (This is done in step 2 of the Playground process)
- Copy the new access token for your `GOOGLE_ACCESS_TOKEN=` in your .env file

- run `sequelize db:migrate`
- run `sequelize db:seed:all`
- run `npm run dev`
 **To edit an accommodation**
- In Postman Provide a login bearer token
- Fill the body with all fields you want to amend
- send a `PATCH` request to `http://localhost:<port>/api/v1/accommodations/{id}/edit`
**To delete an accommodation**
- In Postman Provide a login bearer token
- send a `DELETE` request to `http://localhost:<port>/api/v1/accommodations/{id}/delete`

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
[#169113466](https://www.pivotaltracker.com/story/show/169113466)

#### Screenshots (if appropriate)
Screen Shot 2019-10-17 at 10 53 03
Screen Shot 2019-10-17 at 11 04 43

#### Questions:
N/A

### CODE REVIEW

**Review by Cheza-Dzabala — CHANGES_REQUESTED** [review]
Update the location field to locationId and we're golden.

**Review by alainmateso — APPROVED** [review]
God Job Thomas!

**Review by ngireric123 — APPROVED** [review]

**Comment by nignanthomas** [comment]
> Update the location field to locationId and we're golden.

Changes implemented + Refactoring 👍

**Review by Cheza-Dzabala — APPROVED** [review]

File: src/controllers/accommodationController.js [hunk 1]
```diff
@@ -59,4 +53,40 @@ export default class AccommodationController {
     }
     return responseUtil(res, 200, strings.accommodation.success.RETRIEVED, allAccommodations);
   }
+
+  static async editAccommodation(req, res) {
+    const { body } = req;
```
Comment (nakiwuge) [inline src/controllers/accommodationController.js]: Add a new line here

File: src/controllers/accommodationController.js [hunk 2]
```diff
@@ -59,4 +53,40 @@ export default class AccommodationController {
...
+    if (Object.keys(body).length === 0) {
+      return responseError(res, 400, 'There is nothing to update! Provide some information!');
+    }
+    let images;
+
+    if (req.files) {
+      images = await imageUploader(req, res);
+    }
+    const editAccommodation = (images) ? { ...body, images } : { ...body };
+    const { accommodation } = req;
```
Comment (nakiwuge) [inline src/controllers/accommodationController.js:71]: Add a new line here

File: src/controllers/accommodationController.js [hunk 3]
```diff
@@ -59,4 +53,40 @@ export default class AccommodationController {
...
+  static async editAccommodation(req, res) {
+    const { body } = req;
+    if (Object.keys(body).length === 0) {
+      return responseError(res, 400, 'There is nothing to update! Provide some information!');
+    }
+    let images;
+
+    if (req.files) {
+      images = await imageUploader(req, res);
+    }
```
Comment (nakiwuge) [inline src/controllers/accommodationController.js:68]: Add a new line here

File: src/middlewares/isAccommodationFound.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
+import responseError from '../utils/responseError';
+import strings from '../utils/stringsUtil';
+import models from '../database/models';
+
+const isAccommodationFound = (req, res, next) => {
+  const { id } = req.params;
+  const accommodationId = parseInt(id, 10);
```
Comment (nakiwuge) [inline src/middlewares/isAccommodationFound.js:7]: Add a new line here

File: src/utils/validationChecker.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+    return responseError(
+      res,
+      400,
+      strings.users.error.BAD_SIGNUP_REQUEST,
+      errorMessages
+    );
+  }
+  return next();
+};
+
```
Comment (nakiwuge) [inline src/utils/validationChecker.js]: I do not understand why we have so many validation methods in every feature

**Pull request merged**